src/software/world/robot.cpp

```diff
@@ -3,52 +3,45 @@
 #include <g3log/g3log.hpp>
 
 #include "shared/constants.h"
+#include "software/world/robot_state.h"
 
 Robot::Robot(RobotId id, const Point &position, const Vector &velocity,
              const Angle &orientation, const AngularVelocity &angular_velocity,
-             const Timestamp &timestamp, unsigned int history_duration,
+             const Timestamp &timestamp, unsigned int history_size,
              const std::set<RobotCapabilities::Capability> &capabilities)
-    : id_(id),
-      positions_(history_duration),
-      velocities_(history_duration),
-      orientations_(history_duration),
-      angularVelocities_(history_duration),
-      last_update_timestamps(history_duration),
-      capabilities_(capabilities)
+    : id_(id), states_(history_size), capabilities_(capabilities)
 {
-    addStateToRobotHistory(position, velocity, orientation, angular_velocity, timestamp);
+    if (history_size <= 0)
+    {
+        throw std::invalid_argument("Error: history_size must be greater than 0");
+    }
+
+    updateCurrentState(position, velocity, orientation, angular_velocity, timestamp);
 }
 
-void Robot::updateState(const Point &new_position, const Vector &new_velocity,
-                        const Angle &new_orientation,
-                        const AngularVelocity &new_angular_velocity,
-                        const Timestamp &timestamp)
+void Robot::updateCurrentState(const RobotState &new_state)
 {
-    if (timestamp < lastUpdateTimestamp())
+    if (!states_.empty() && new_state.timestamp() < lastUpdateTimestamp())
     {
         throw std::invalid_argument(
-            "Error: State of robot is updating times from the past");
+            "Error: Trying to update robot state using a state older then the current state");
     }
 
-    addStateToRobotHistory(new_position, new_velocity, new_orientation,
-                           new_angular_velocity, timestamp);
+    states_.push_front(new_state);
 }
 
-void Robot::updateState(const Robot &new_robot_data)
+void Robot::updateCurrentState(const Point &new_position, const Vector &new_velocity,
```

Comment:
I think we don't really need this function? It kinda defeats the point of having the `State` class since it just exposes all the same members.

Reply:
@garethellis0 @scveloso @jonathanlew thoughts on this?

Reply:
I'm fine with this being public.

Reply:
Agree, I think caller should be using a RobotState if caller has those fields anyway

Reply:
I'm OK with it being public

Reply:
removed this one in favour of just having updateState(RobotState state)